lib/reader.js

```diff
@@ -39,33 +39,35 @@ var readAll = function (filePath, frontMatterConfig) {
     return { html, frontMatter };
 }
 
-var readFromUrl = function (postUrl) {
-    return new Promise((resolve, reject) => {
-        request.get(postUrl, function (error, response, body) {
-            if (error) {
-                reject(error);
-            }
-
-            let $ = cheerio.load(response.body);
-            $('.section-divider').remove();
-
-            const html = $('.postArticle-content').html() || '';
-
-            const title = $(".graf--title").text();
-            const subtitle = $("meta[name='description']").attr("content");
-            const date = $("meta[property='article:published_time']").attr("content");
-            const canonical = $("link[rel='canonical']").attr("href");
-            const slug = canonical ? url.parse(canonical).path : '';
-
-            const tags = [];
-            $(".js-postTags li").each((i, e) => {
-                tags.push($(e).text());
-            });
-
-            const frontMatter = generateFrontMatter(title, subtitle, date, slug, tags);
-            resolve({ html, frontMatter, title });
-        });
-    });
+var readFromUrl = async function (postUrl) {
+    try {
+        const response = await fetch(postUrl);
+        const body = await response.text();
+
+        const $ = cheerio.load(body);
+
+        $('.section-divider').remove();
+
+        const html = $('.postArticle-content').html() || '';
+        const title = $(".graf--title").text();
+        const subtitle = $("meta[name='description']").attr("content");
+        const date = $("meta[property='article:published_time']").attr("content");
+        const canonical = $("link[rel='canonical']").attr("href");
+        const slug = canonical ? url.parse(canonical).path : '';
+
+        const tags = [];
+        $(".js-postTags li").each((i, e) => tags.push($(e).text()));
+
+        const frontMatter = generateFrontMatter(title, subtitle, date, slug, tags);
+
+        return {
+            html,
+            title,
+            frontMatter,
+        };
+    } catch (error) {
+        throw error;

```

Comment:
What's the point when we are catching and throwing the same error? Either catch and handle it else remove the try block.